### PR TITLE
Bug 1876332 - Refactoring out Translations Controller and Interactor

### PR DESCRIPTION
**Pull request by ohall-m**
This patch refactors out the translations controller and interactor to transition fully to a redux pattern.

Leaving the translations use cases for reference browser usage.

### Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [X] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1876332

### CODE REVIEW

**Comment by ohall-m** [comment]
Didn't remove the use cases because reference browser is using them.

File: fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt [hunk 1]
```diff
@@ -125,14 +115,7 @@ class TranslationsDialogFragment : BottomSheetDialogFragment() {
...
                                     },
-                                    onTranslateButtonClick = {
-                                        interactor.onTranslate(
-                                            tabId = args.sessionId,
-                                            fromLanguage = null,
-                                            toLanguage = null,
-                                            null,
-                                        )
-                                    },
+                                    onTranslateButtonClick = {},
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt:118]: Did you intend on only removing the Interactor/Controller in this PR?

Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt:118]: That's right, I was also going to do [use case itself too](https://searchfox.org/mozilla-mobile/rev/99576d6e69eac8ee1bf6c8a654d5fa96a6f6bf84/firefox-android/android-components/components/feature/session/src/main/java/mozilla/components/feature/session/SessionUseCases.kt#469,499), but forgot reference browser was using the [use cases](https://searchfox.org/mozilla-mobile/rev/063c6bcf58858b26ebdc8cd0c8f2c658d82061da/firefox-android/android-components/samples/browser/src/main/java/org/mozilla/samples/browser/DefaultComponents.kt#326-338,344).

Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogFragment.kt:118]: (Mostly trying to [revert to this state](https://github.com/mozilla-mobile/firefox-android/pull/5148/files), sans getting rid of the id passing.)

**Review by gabrielluong — APPROVED** [review]

**Comment by MatthewTighe** [comment]
> Didn't remove the use cases because reference browser is using them.

Will we plan to down the road, or just leave them in for reference browser?

**Comment by ohall-m** [comment]
>Will we plan to down the road, or just leave them in for reference browser? 

I'm inclined to keep the use cases in-tree for reference browser because that seems to be the overall pattern that was used  over there. I like having the reference browser for testing and debugging down the road. 

Otherwise, I think the effort to benefit ratio might not support refactoring reference browser and would be inclined just to remove the menu options along with the use case.

**Review by iorgamgabriel — APPROVED** [review]

**Pull request merged**